js/src/ui/Form/AddressSelect/addressSelect.js

```diff
@@ -15,262 +15,576 @@
-    const addresses = Object.keys(entries).sort();
+  setValues (props = this.props) {
+    const { accounts = {}, contracts = {}, contacts = {}, wallets = {} } = props;
 
-    this.setState({ autocompleteEntries, entries, addresses, value });
-  }
+    const accountsN = Object.keys(accounts).length;
+    const contractsN = Object.keys(contracts).length;
+    const contactsN = Object.keys(contacts).length;
+    const walletsN = Object.keys(wallets).length;
 
-  componentWillReceiveProps (newProps) {
-    if (newProps.value !== this.props.value) {
-      this.setState({ value: newProps.value });
+    if (accountsN + contractsN + contactsN + walletsN === 0) {
+      return;
     }
+
+    this.values = [
+      {
+        label: 'accounts',
+        values: [].concat(
+          Object.values(wallets),
+          Object.values(accounts)
+        )
+      },
+      {
+        label: 'contacts',
+        values: Object.values(contacts)
+      },
+      {
+        label: 'contracts',
+        values: Object.values(contracts)
+      }
+    ];
+
+    this.handleChange();
   }
 
   render () {
-    const { allowInput, disabled, error, hint, label } = this.props;
-    const { autocompleteEntries, value } = this.state;
+    const input = this.renderInput();
+    const content = this.renderContent();
 
-    const searchText = this.getSearchText();
-    const icon = this.renderIdentityIcon(value);
+    const classes = [ styles.main ];
 
     return (
-      <div className={ styles.container }>
-        <AutoComplete
-          className={ !icon ? '' : styles.paddedInput }
-          disabled={ disabled }
-          label={ label }
-          hint={ hint ? `search for ${hint}` : 'search for an address' }
-          error={ error }
-          onChange={ this.onChange }
-          onBlur={ this.onBlur }
-          onUpdateInput={ allowInput && this.onUpdateInput }
-          value={ searchText }
-          filter={ this.handleFilter }
-          entries={ autocompleteEntries }
-          entry={ this.getEntry() || {} }
-          renderItem={ this.renderItem }
-        />
-        { icon }
+      <div
+        className={ classes.join(' ') }
+        onBlur={ this.handleMainBlur }
+        onClick={ this.handleFocus }
+        onFocus={ this.handleMainFocus }
+        onKeyDown={ this.handleInputAddresKeydown }
+        ref='inputAddress'
+        tabIndex={ 0 }
+      >
+        { input }
+        { content }
       </div>
     );
   }
 
-  renderIdentityIcon (inputValue) {
-    const { error, value, label } = this.props;
-
-    if (error || !inputValue || value.length !== 42) {
-      return null;
-    }
-
-    const classes = [ styles.icon ];
+  renderInput () {
+    const { focused } = this.state;
+    const { accountsInfo, disabled, error, hint, label, value } = this.props;
+
+    const input = (
+      <InputAddress
+        accountsInfo={ accountsInfo }
+        allowCopy={ false }
+        disabled
```

Comment:
General comment. `readOnly` would be appropriate here, [but is not yet supported on `<Input>`](https://github.com/callemall/material-ui/pull/2289). [`disabled` means "the form control is not available for interaction"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#attr-disabled).